crates/bevy_ecs/src/schedule/graph/graph_map.rs

```diff
@@ -249,28 +248,65 @@ where
     }
 
     /// Return an iterator over all edges of the graph with their weight in arbitrary order.
-    pub fn all_edges(&self) -> impl ExactSizeIterator<Item = (NodeId, NodeId)> + '_ {
-        self.edges.iter().copied().map(CompactNodeIdPair::load)
+    pub fn all_edges(&self) -> impl ExactSizeIterator<Item = (N, N)> + '_ {
+        self.edges.iter().copied().map(N::Pair::unwrap)
     }
 
-    pub(crate) fn to_index(&self, ix: NodeId) -> usize {
+    pub(crate) fn to_index(&self, ix: N) -> usize {
         self.nodes.get_index_of(&ix).unwrap()
     }
+
+    /// Converts from one [`GraphNodeId`] type to another. If the conversion fails,
+    /// it returns the error from the target type's [`TryFrom`] implementation.
+    ///
+    /// # Errors
+    ///
+    /// If the conversion fails, it returns an error of type `T::Error`.
+    pub fn try_into<T: GraphNodeId + TryFrom<N>>(self) -> Result<Graph<DIRECTED, T, S>, T::Error>
+    where
+        S: Default,
+    {
+        let nodes = self
+            .nodes
+            .into_iter()
+            .map(|(k, v)| {
+                Ok((
+                    k.try_into()?,
+                    v.into_iter()
+                        .map(|v| {
+                            let (id, dir) = v.unwrap();
+                            Ok(T::Directed::new(id.try_into()?, dir))
+                        })
+                        .collect::<Result<Vec<T::Directed>, T::Error>>()?,
+                ))
+            })
+            .collect::<Result<IndexMap<T, Vec<T::Directed>, S>, T::Error>>()?;
```

Comment:
This will do something weird if multiple keys `try_into()` the same value.  We don't need to do that, so it's not worth supporting, but it might be worth noting in the doc comment that it's a requirement.

Reply:
Note added in the function docs.